src/python/pants/backend/python/rules/coverage_integration_test.py

```diff
@@ -10,139 +10,139 @@
 
 
 class CoverageIntegrationTest(PantsRunIntegrationTest):
-    def test_coverage(self) -> None:
-        with temporary_dir(root_dir=get_buildroot()) as tmpdir:
-            tmpdir_relative = Path(tmpdir).relative_to(get_buildroot())
-            src_root = Path(tmpdir, "src", "python", "project")
-            src_root.mkdir(parents=True)
-            (src_root / "__init__.py").touch()
-
-            # Set up the source files. Only `lib.py` will actually be tested, but we still expect
-            # `random.py` to show up in the final report correctly.
-            (src_root / "lib.py").write_text(
-                dedent(
-                    """\
-                    def add(x, y):
-                        return x + y
-
-                    def subtract(x, y):
-                        return x - y
-
-                    def multiply(x, y):
-                        return x * y
-                    """
-                )
+    def _prepare_sources(self, tmpdir, build_root):
```

Comment:
Fine to fix in a followup.

```suggestion
    def _prepare_sources(self, tmpdir: str, build_root: str) -> str:
```

Reply:
Also, in a followup, the code might be simpler if you remove `build_root` as a param. `get_buildroot()` is a singleton, and it's also extremely cheap to call. So you can inline it on the next line to avoid having extra variables and an extra param.